### DISCUSSION TITLE
How to get hash encoded in something else than hex?

### DISCUSSION THREAD

**Discussion by jdorn-eu** [wierkstudio/ciphereditor#22] [Help] [answered]
On cryptii.com I could do something like this:
<img width="1106" alt="スクリーンショット 2022-12-07 19 48 30" src="https://user-images.githubusercontent.com/3247063/206269573-30f43dbf-4dd0-41d1-8b64-3a019e525da6.png">

But now it doesn't work anymore: [Blueprint link 🔗](https://ciphereditor.com/share#blueprint=eyJ0eXBlIjoiYmx1ZXByaW50IiwicHJvZ3JhbSI6eyJ0eXBlIjoicHJvZ3JhbSIsImxhYmVsIjoiUHJvZ3JhbSIsIm9mZnNldCI6eyJ4IjowLCJ5IjowfSwiZnJhbWUiOnsieCI6LTE2MCwieSI6LTk2LCJ3aWR0aCI6MzIwLCJoZWlnaHQiOjE5Mn0sImNoaWxkcmVuIjpbeyJ0eXBlIjoib3BlcmF0aW9uIiwibmFtZSI6IkBjaXBoZXJlZGl0b3IvZXh0ZW5zaW9uLWhhc2gvaGFzaCIsImV4dGVuc2lvblVybCI6Imh0dHBzOi8vY2RuLmNpcGhlcmVkaXRvci5jb20vZXh0ZW5zaW9ucy9AY2lwaGVyZWRpdG9yL2V4dGVuc2lvbi1oYXNoLzEuMC4wLWFscGhhLjEvZXh0ZW5zaW9uLmpzIiwicHJpb3JpdHlDb250cm9sTmFtZXMiOlsibWVzc2FnZSIsImFsZ29yaXRobSIsImhhc2giXSwiZnJhbWUiOnsieCI6LTE2MCwieSI6LTk2LCJ3aWR0aCI6MzIwLCJoZWlnaHQiOjQ0Nn0sImNvbnRyb2xzIjp7Im1lc3NhZ2UiOnsidmFsdWUiOiJoZWxsbyIsInZpc2liaWxpdHkiOiJleHBhbmRlZCJ9LCJhbGdvcml0aG0iOnsidmFsdWUiOiJzaGEyNTYiLCJ2aXNpYmlsaXR5IjoiZXhwYW5kZWQifSwiaGFzaCI6eyJpZCI6IjUiLCJ2YWx1ZSI6IjJjZjI0ZGJhNWZiMGEzMGUyNmU4M2IyYWM1YjllMjllMWIxNjFlNWMxZmE3NDI1ZTczMDQzMzYyOTM4Yjk4MjQiLCJ2aXNpYmlsaXR5IjoiZXhwYW5kZWQifX19LHsidHlwZSI6Im9wZXJhdGlvbiIsIm5hbWUiOiJAY2lwaGVyZWRpdG9yL2V4dGVuc2lvbi1lc3NlbnRpYWxzL2JpbmFyeS10by10ZXh0IiwiZXh0ZW5zaW9uVXJsIjoiaHR0cHM6Ly9jZG4uY2lwaGVyZWRpdG9yLmNvbS9leHRlbnNpb25zL0BjaXBoZXJlZGl0b3IvZXh0ZW5zaW9uLWVzc2VudGlhbHMvMS4wLjAtYWxwaGEuMS9leHRlbnNpb24uanMiLCJwcmlvcml0eUNvbnRyb2xOYW1lcyI6WyJkYXRhIiwiYWxwaGFiZXQiLCJwYWRkaW5nIiwiZW5jb2RlZERhdGEiXSwiZnJhbWUiOnsieCI6LTEyMCwieSI6LTU2LCJ3aWR0aCI6MzIwLCJoZWlnaHQiOjYxNH0sImNvbnRyb2xzIjp7ImRhdGEiOnsiaWQiOiI3IiwidmFsdWUiOiIyY2YyNGRiYTVmYjBhMzBlMjZlODNiMmFjNWI5ZTI5ZTFiMTYxZTVjMWZhNzQyNWU3MzA0MzM2MjkzOGI5ODI0IiwidmlzaWJpbGl0eSI6ImV4cGFuZGVkIn0sImFscGhhYmV0Ijp7InZpc2liaWxpdHkiOiJleHBhbmRlZCJ9LCJwYWRkaW5nIjp7InZpc2liaWxpdHkiOiJleHBhbmRlZCJ9LCJlbmNvZGVkRGF0YSI6eyJ2YWx1ZSI6Ik1tTm1NalJrWW1FMVptSXdZVE13WlRJMlpUZ3pZakpoWXpWaU9XVXlPV1V4WWpFMk1XVTFZekZtWVRjME1qVmxOek13TkRNek5qSTVNemhpT1RneU5BPT0iLCJ2aXNpYmlsaXR5IjoiZXhwYW5kZWQifX19LHsidHlwZSI6InZhcmlhYmxlIiwiYXR0YWNobWVudHMiOlsiNSIsIjciXX1dfX0)

I think the output of the hash function is interpreted as text instead of bytes, but is there an option to change that?

(Also on that note: base64url gives me the same output as base64, it seems to be broken)

**Comment by ffraenz** [comment, marked as answer]
Hi there! The hash function operation outputs text when text is sent to it. When you want to get bytes out of it, send in bytes. This behavior is not documented, yet. This might be changed in the future to make it more intuitive (e.g. by adding a format control). In [this blueprint](https://ciphereditor.com/share#blueprint=eyJ0eXBlIjoiYmx1ZXByaW50IiwicHJvZ3JhbSI6eyJ0eXBlIjoicHJvZ3JhbSIsImxhYmVsIjoiUHJvZ3JhbSIsIm9mZnNldCI6eyJ4Ijo2MCwieSI6MTY4fSwiZnJhbWUiOnsieCI6LTE2MCwieSI6LTk2LCJ3aWR0aCI6MzIwLCJoZWlnaHQiOjE5Mn0sImNoaWxkcmVuIjpbeyJ0eXBlIjoib3BlcmF0aW9uIiwibmFtZSI6IkBjaXBoZXJlZGl0b3IvZXh0ZW5zaW9uLWhhc2gvaGFzaCIsImV4dGVuc2lvblVybCI6Imh0dHBzOi8vY2RuLmNpcGhlcmVkaXRvci5jb20vZXh0ZW5zaW9ucy9AY2lwaGVyZWRpdG9yL2V4dGVuc2lvbi1oYXNoLzEuMC4wLWFscGhhLjEvZXh0ZW5zaW9uLmpzIiwicHJpb3JpdHlDb250cm9sTmFtZXMiOlsibWVzc2FnZSIsImFsZ29yaXRobSIsImhhc2giXSwiZnJhbWUiOnsieCI6LTE2MCwieSI6LTk2LCJ3aWR0aCI6MzIwLCJoZWlnaHQiOjIzOH0sImNvbnRyb2xzIjp7Im1lc3NhZ2UiOnsiaWQiOiI0IiwidmFsdWUiOnsidHlwZSI6ImJ5dGVzIiwiZGF0YSI6ImFHVnNiRzhnZDI5eWJHUT0ifX0sImFsZ29yaXRobSI6eyJ2YWx1ZSI6InNoYTI1NiIsInZpc2liaWxpdHkiOiJleHBhbmRlZCJ9LCJoYXNoIjp7ImlkIjoiNiIsInZhbHVlIjp7InR5cGUiOiJieXRlcyIsImRhdGEiOiJ1VTBudVpOTlBnaWxMbExYMm4ycitzU0U3K042VTREdWtJajNyT0x2emVrPSJ9fX19LHsidHlwZSI6Im9wZXJhdGlvbiIsIm5hbWUiOiJAY2lwaGVyZWRpdG9yL2V4dGVuc2lvbi1lc3NlbnRpYWxzL2JpbmFyeS10by10ZXh0IiwiZXh0ZW5zaW9uVXJsIjoiaHR0cHM6Ly9jZG4uY2lwaGVyZWRpdG9yLmNvbS9leHRlbnNpb25zL0BjaXBoZXJlZGl0b3IvZXh0ZW5zaW9uLWVzc2VudGlhbHMvMS4wLjAtYWxwaGEuMS9leHRlbnNpb24uanMiLCJwcmlvcml0eUNvbnRyb2xOYW1lcyI6WyJkYXRhIiwiYWxwaGFiZXQiLCJwYWRkaW5nIiwiZW5jb2RlZERhdGEiXSwiZnJhbWUiOnsieCI6LTE1NywieSI6MTg4LCJ3aWR0aCI6MzIwLCJoZWlnaHQiOjQzOH0sImNvbnRyb2xzIjp7ImRhdGEiOnsiaWQiOiI4IiwidmFsdWUiOnsidHlwZSI6ImJ5dGVzIiwiZGF0YSI6InVVMG51Wk5OUGdpbExsTFgybjJyK3NTRTcrTjZVNER1a0lqM3JPTHZ6ZWs9In19LCJhbHBoYWJldCI6eyJ2aXNpYmlsaXR5IjoiZXhwYW5kZWQifSwicGFkZGluZyI6eyJ2aXNpYmlsaXR5IjoiZXhwYW5kZWQifSwiZW5jb2RlZERhdGEiOnsidmFsdWUiOiJ1VTBudVpOTlBnaWxMbExYMm4ycitzU0U3K042VTREdWtJajNyT0x2emVrPSIsInZpc2liaWxpdHkiOiJleHBhbmRlZCJ9fX0seyJ0eXBlIjoib3BlcmF0aW9uIiwibmFtZSI6IkBjaXBoZXJlZGl0b3IvZXh0ZW5zaW9uLWNvZGVwYWdlcy90ZXh0LWVuY29kZXIiLCJleHRlbnNpb25VcmwiOiJodHRwczovL2Nkbi5jaXBoZXJlZGl0b3IuY29tL2V4dGVuc2lvbnMvQGNpcGhlcmVkaXRvci9leHRlbnNpb24tY29kZXBhZ2VzLzEuMC4wLWFscGhhLjEvZXh0ZW5zaW9uLmpzIiwicHJpb3JpdHlDb250cm9sTmFtZXMiOlsidGV4dCIsImVuY29kaW5nIiwiYnl0ZXMiXSwiZnJhbWUiOnsieCI6LTE2MSwieSI6LTQwNywid2lkdGgiOjMyMCwiaGVpZ2h0IjoyNzB9LCJjb250cm9scyI6eyJ0ZXh0Ijp7InZhbHVlIjoiaGVsbG8gd29ybGQiLCJ2aXNpYmlsaXR5IjoiZXhwYW5kZWQifSwiYnl0ZXMiOnsiaWQiOiIxNiIsInZhbHVlIjp7InR5cGUiOiJieXRlcyIsImRhdGEiOiJhR1ZzYkc4Z2QyOXliR1E9In19fX0seyJ0eXBlIjoidmFyaWFibGUiLCJhdHRhY2htZW50cyI6WyI2IiwiOCJdfSx7InR5cGUiOiJ2YXJpYWJsZSIsImF0dGFjaG1lbnRzIjpbIjE2IiwiNCJdfV19fQ) I placed a text encoder operation before the hash function to get bytes out of the latter.

The base64url alphabet only replaces the characters `+` and `/` from the original alphabet by `-` and `_`. As these don't appear in your output, they appear to be the same.

Let me know if you have further questions or suggestions!

**Reply by jdorn-eu** [reply]
Thanks very clear now!
